pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -103,6 +103,19 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 					break
 				}
 			}
+			// If RevisionConditionResourcesAvailable is false, with reason being "PodInitializing",
+			// check to see if the secrets are fetchable
+			if cond := rev.Status.GetCondition(v1alpha1.RevisionConditionResourcesAvailable); cond != nil && cond.Status == corev1.ConditionFalse && cond.Reason == "PodInitializing" {
+				for _, volume := range rev.Spec.Volumes {
+					if secret := volume.VolumeSource.Secret; secret != nil {
+						_, err := c.secretLister.Secrets(rev.Namespace).Get(secret.SecretName)
+						if err != nil {
+							rev.Status.MarkSecretUnavailable(secret.SecretName, err.Error())
+							break
```

Comment:
now that we are doing the work, maybe rather than breaking out of the loop on the first missing secret, we loop over and have an aggregate report for all missing secrets on the status?

Reply:
@nimakaviani makes sense! I'll add some logs for all errors fetching secrets and not break out of the loop.